workspace/src/main/kotlin/com/cosmotech/workspace/service/WorkspaceServiceImpl.kt

```diff
@@ -437,7 +440,7 @@ internal class WorkspaceServiceImpl(
         workspaceRepository.findByIdOrNull(workspaceId)
             ?: throw CsmResourceNotFoundException("Workspace $workspaceId does not exist!")
     csmRbac.verify(workspace.getRbac(), requiredPermission)
-    return workspace
+    return checkReadSecurity(workspace)
```

Comment:
Same remark, there is no need to do that here as we just need to check read permission in 2-3 methods.
Plus getVerified* methods are utilities methods for dev, there not exposed to users.